### DISCUSSION TITLE
Can we get the intramolecular hydrogen bond number from molecular conformers?

### DISCUSSION THREAD

**Discussion by josiehong** [rdkit/rdkit#5829] [Q&A] [answered]
Hi, 

I am working on intramolecular hydrogen bonds (IMHB). Assuming the conformer is known, I am curious whether we could get the number of IMHBs. 

By the way, I noticed that the number of hydrogen bond acceptors and donors can be calculated by `rdkit.Chem.Lipinski` module. Maybe this module is helpful. 

Thanks!

**Comment by greglandrum** [comment]
Hi @JosieHong, sorry for the very slow reply to this one.
Are you still interested in an answer to this question?

**Reply by josiehong** [reply]
Yes, I am still interested. Do you have any idea about that?

**Reply by greglandrum** [reply, marked as answer]
Here's a function which loops through all the atoms in a molecule and identifies short distances between hydrogens attached to polar atoms and other polar atoms. As written, only N and O are considered as polar atoms, but you can change this if you want.

```
def find_intramolecular_hbonds(mol,confId=-1,eligibleAtoms=[7,8],distTol=2.5):
    '''
    eligibleAtoms is the list of atomic numbers of eligible H-bond donors or acceptors
    distTol is the maximum accepted distance for an H bond
    '''
    res = []
    conf = mol.GetConformer(confId)
    for i in range(mol.GetNumAtoms()):
        atomi = mol.GetAtomWithIdx(i)
        # is it H?
        if atomi.GetAtomicNum()==1:
            if atomi.GetDegree() != 1:
                continue
            nbr = atomi.GetNeighbors()[0]
            if nbr.GetAtomicNum() not in eligibleAtoms:
                continue
            # loop over all other atoms except ones we're bound to and other Hs:
            for j in range(mol.GetNumAtoms()):
                if j==i:
                    continue
                atomj = mol.GetAtomWithIdx(j)
                if atomj.GetAtomicNum() not in eligibleAtoms or mol.GetBondBetweenAtoms(i,j):
                    continue
                dist = (conf.GetAtomPosition(i)- conf.GetAtomPosition(j)).Length()
                if dist<distTol:
                    res.append((i,j,dist))
    return res
```
And here's a gist with a demo of how to use the function;
https://gist.github.com/greglandrum/b93b1b25a4974e7d5e6908088b5fa58e

**Reply by manajit-das** [reply]
When I try for o-hydroxy benzoic acid 'OC1=CC=CC=C1C(O)=O' it doesn't result the desired output, as one H-bond is expected in this molecule. The carbonyl and the hydroxyl group is not close in the generated conformer. What to do in such case?
Thanks.
